autotest/ogr/ogr_gpkg.py

```diff
@@ -4146,3 +4148,38 @@ def test_ogr_gpkg_datetime_timezones():
     ds = None
 
     gdal.Unlink(filename)
+
+
+###############################################################################
+# Test AbortSQL
+
+def test_abort_sql():
+
+    filename = '/vsimem/test_ogr_gpkg_abort_sql.gpkg'
+    ds = ogr.GetDriverByName('GPKG').CreateDataSource(filename)
+    ds.CreateLayer('test')
+    ds = None
+
+    ds = ogr.Open(filename, update = 1)
+
+    def abortAfterDelay():
+        print("Aborting SQL...")
+        ds.AbortSQL()
+
+    t = threading.Timer(0.5, abortAfterDelay)
+    t.start()
+
+    start = time.time()
+
+    # Long running query
+    ds.ExecuteSQL("""
```

Comment:
I wouldn't be surprised to see failure on some CI targets that ship with not so recent SQLite, so you'll probably have to conditionalize that test

Reply:
I think I need a little help here, ti looks like there are some unrelated failures, but let's wait for the new results after my last commits.